gateway/coprocess.go

```diff
@@ -362,11 +362,15 @@ func (m *CoProcessMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Requ
 		if returnObject.Session == nil || token == "" {
 			authHeaderValue := r.Header.Get(m.Spec.Auth.AuthHeaderName)
 			AuthFailed(m, r, authHeaderValue)
-			return errors.New("Key not authorised"), 403
+			return errors.New(http.StatusText(http.StatusForbidden)), http.StatusForbidden
```

Comment:
Not sure `http.StatusText` is the best approach here. It will return "Forbidden". I'd say ___Key not authorised___ is way more descriptive and easier to debug.

Reply:
I think that this is a security leak and introduces an attack vector. If I hit the api with a bad key i will know that the server responding is Tyk.

Also, i've tweaked it a bit to use NotAuthorized & Forbidden. PTAL